src/lib.rs

```diff
@@ -670,6 +686,16 @@ impl Options {
     pub fn frequency(&self) -> u32 {
         self.frequency.unwrap_or(997)
     }
+
+    #[cfg(target_os = "linux")]
+    fn compression_level(&self) -> Option<u8> {
+        self.compression_level
+    }
+
+    #[cfg(not(target_os = "linux"))]
+    fn compression_level(&self) -> Option<u8> {
+        None
+    }
```

Comment:
Would prefer to keep this in the caller. I think this should work?

```rust
#[cfg(target_os = "linux")]
let compression = opts.compression_level;
#[cfg(not(target_os = "linux"))]
let compression = None;
```

Reply:
sure